### ISSUE TITLE
Internal exception when executing rules with no strings defined

### ISSUE THREAD

**Issue by ITAYC0HEN** [CERT-Polska/mquery#36]
**Description**

When no strings are provided in the Yara rule, mquery will get an internal server error and a python exception.

Some Yara rules do not contain any strings, As can be seen in the official Yara documentation and in the following examples:
```
rule FileSizeExample
{
    condition:
       filesize > 200KB
}
```

```
rule IsPE
{
  condition:
     // MZ signature at offset 0 and ...
     uint16(0) == 0x5A4D and
     // ... PE signature at offset stored in MZ header at 0x3C
     uint32(uint32(0x3C)) == 0x00004550
}
```

```
import "hash"

rule test
{
    condition:
        hash.sha256(0, filesize) == "e9455dcfaa7571ef460913879ad8c562d4abb374316bdf242b2459c7f85519e6"
}
```

**Cause to the problem**
From a quick investigation, it seems like the object `job_obj` is passed to `redis.hmset(..., job_obj)` with a NoneType value for the `parsed` key in the dictionary.

https://github.com/CERT-Polska/mquery/blob/5001934096c8522b88083eaf812c80ca59c3d466/src/app.py#L91-L122

There is no validation check for the value before passing it, which isn't a good practice. 
`redis-py no longer accepts NoneType  objects: https://github.com/andymccurdy/redis-py/issues/1071#issuecomment-439199774

**Expected behavior**
The query should be executed without any problem

**Comment by ITAYC0HEN** [comment]
The same happens when strings are defined but are not used:

```
rule test
{
    strings:
        $a = "test string"

    condition:
       filesize > 200KB
}
```

**Comment by msm-code** [comment]
This is easy to fix, but the system should warn user in this case (see also issue about known limitations). Doing a query on all files will take a long time on bigger datasets (which may or may not be what the user wanted).

**Comment by msm-code** [comment]
Fixed during the daemon rewrite (#119 )
